src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs

```diff
@@ -13,27 +14,35 @@ internal class LinuxAppServiceEventGenerator : LinuxEventGenerator
         private readonly Action<string> _writeEvent;
         private readonly HostNameProvider _hostNameProvider;
         private readonly IOptions<FunctionsHostingConfigOptions> _functionsHostingConfigOptions;
-        // Loggers
-        private readonly Lazy<LinuxAppServiceFileLogger> _functionsExecutionEventsCategoryLogger;
-        private readonly Lazy<LinuxAppServiceFileLogger> _functionsLogsCategoryLogger;
-        private readonly Lazy<LinuxAppServiceFileLogger> _functionsMetricsCategoryLogger;
-        private readonly Lazy<LinuxAppServiceFileLogger> _functionsDetailsCategoryLogger;
+        private readonly Lazy<ILinuxAppServiceFileLogger> _functionsExecutionEventsCategoryLogger;
+        private readonly Lazy<ILinuxAppServiceFileLogger> _functionsLogsCategoryLogger;
+        private readonly Lazy<ILinuxAppServiceFileLogger> _functionsMetricsCategoryLogger;
+        private readonly Lazy<ILinuxAppServiceFileLogger> _functionsDetailsCategoryLogger;
 
         public LinuxAppServiceEventGenerator(
-            LinuxAppServiceFileLoggerFactory loggerFactory,
+            ILinuxAppServiceFileLoggerFactory loggerFactory,
             HostNameProvider hostNameProvider,
             IOptions<FunctionsHostingConfigOptions> functionsHostingConfigOptions,
             Action<string> writeEvent = null)
         {
             _writeEvent = writeEvent ?? WriteEvent;
             _hostNameProvider = hostNameProvider ?? throw new ArgumentNullException(nameof(hostNameProvider));
             _functionsHostingConfigOptions = functionsHostingConfigOptions;
-            _functionsExecutionEventsCategoryLogger = loggerFactory.Create(FunctionsExecutionEventsCategory, backoffEnabled: !functionsHostingConfigOptions.Value.DisableLinuxAppServiceLogBackoff);
-            _functionsLogsCategoryLogger = loggerFactory.Create(FunctionsLogsCategory, backoffEnabled: false);
-            _functionsMetricsCategoryLogger = loggerFactory.Create(FunctionsMetricsCategory, backoffEnabled: false);
-            _functionsDetailsCategoryLogger = loggerFactory.Create(FunctionsDetailsCategory, backoffEnabled: false);
+            _functionsExecutionEventsCategoryLogger = loggerFactory.Create(FunctionsExecutionEventsCategory, functionsHostingConfigOptions.Value.DisableLinuxAppServiceLogBackoff);
```

Comment:
Added back backoffEnabled annotation. Flag is to disable the backoff.  So negate the flag to supply the execution logger constructor.   Annotation makes the behavior clearer.

Reply:
hmmm it was more confusing to read !<flag> --> do not define variables if its only use is in constructor